lib/tests/common.go

```diff
@@ -0,0 +1,144 @@
+package test
+
+import (
+	"archive/tar"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/appc/docker2aci/lib/internal/typesV2"
+)
+
+type Layer map[*tar.Header][]byte
```

Comment:
What I'm doing here is trying to work around the fact that Go doesn't have tuples. I want to store a list of pairs, where each pair is a tar header and the file contents. Go's maps kinda let me do this, by being a way to define a set of pairs (where each key and its value is the pair).

I could specify a struct that holds a `tar.Header` and a `[]byte`, and have a layer be a slice of these structs, but that also would allow duplicate tar headers, so there's no benefit there over the current approach.

I don't think using layer hashes as the map keys here makes sense, as I don't know the hashes at this point in time and the keys here represent a given file in the layer, not the layer itself.

To truly prevent duplicates perhaps I could go with the slice of structs approach and have a helper function to search the slice each time to make sure that I'm not adding duplicates, but that seems rather heavy handed for hardcoded tests.